### PR TITLE
crimson: clang related cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/crimson/os/seastore/root_block.h [hunk 1]
```diff
@@ -97,10 +97,12 @@ struct RootBlock : CachedExtent {
 
   complete_load_ertr::future<> complete_load() final {
     assert(0 == "Root is only written via deltas");
+    throw std::logic_error("Root is only written via deltas");
```
Comment (rzarzynski) [inline src/crimson/os/seastore/root_block.h:100]: How about `ceph_abort_msg()`?

Comment (tchaikov) [reply src/crimson/os/seastore/root_block.h]: @rzarzynski thanks! updated and repushed.

**Comment by tchaikov** [comment]
```
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 0.155065/0.17634  => rejected
16:37:35 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: cpu_cycles_per_op: (or (less) (near 0.05)):: 1296307.4921875/2172571.0  => accepted
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 39.0/45.0  => rejected
16:37:35 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 2.00)):: 23.7978/17.6163  => accepted
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.401711/0.353254  => rejected
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 0.155066/0.17635  => rejected
16:37:35 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: cpu_cycles_per_op: (or (less) (near 0.05)):: 1296307.4921875/2172571.0  => accepted
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 39.0/45.0  => rejected
16:37:35 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 2.00)):: 24.4404/18.4752  => accepted
16:37:35 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.402993/0.354348  => rejected
16:37:35 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 44.7528/86.8437  => rejected
16:37:35 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: cpu_cycles_per_op: (or (less) (near 0.05)):: 290763.88970541937/227159.18883449657  => rejected
16:37:35 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 11456.0/22231.0  => rejected
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 2.00)):: 17240.2/22207.7  => accepted
16:37:35 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.00139244/0.000715927  => rejected
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 49.6191/48.9776  => accepted
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: cpu_cycles_per_op: (or (less) (near 0.05)):: 262244.3240795735/371289.3747221589  => accepted
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 12702.0/12538.0  => accepted
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 2.00)):: 19560.7/18528.7  => accepted
16:37:35 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.0012557/0.00127211  => accepted
16:37:35 - WARNING  - cbt      - 10 tests failed out of 20
```

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**